Week-3/Assignment-1/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @param fn - a function which is to be memoized
+ * @returns - a function which accepts n number of params & returns memoized value from the cache store
+ */
+function memoize(fn) {
+    const cache = new Map();
+    return function (...args) {
+        const key = args.toString();
+        if (cache.has(key)) {
+            return cache.get(key);
+        }
+        cache.set(key, fn(...args));
+        return cache.get(key);
+    };
+}
+
+/**
+ * @param fn - a function for which the performance to be captured
+ */
+function timeLog(fn) {
+    console.time();
+    console.log(fn());
+    console.timeEnd();
+}
+
+/**
+ * @param args - array of n numbers
+ * @returns - a sum of all the inputs
+ */
+function add(...args) {
+    return args.reduce((prevArg, currArg) => Number(prevArg) + Number(currArg), 0);
```

Comment:
In reduce method, the first argument to the callback function is called **accumulator**, which in your case should have been the total sum of previous elements and not the prevArg. Please correct it.

Reply:
Thanks @sairamin, I've now updated it, please review & merge